app/controllers/templates_controller.rb

```diff
@@ -0,0 +1,13 @@
+class TemplatesController < ApplicationController
+  before_action :authenticate_user!
+
+  def index
+    @templates = Template.all
+  end
+
+  def show
+    @template = Template.find_by(id: params[:id])
```

Comment:
You could use find rather than find_by here and let rails server a 404 if it's not found.